pkg/trace/api/evp_intake_proxy_test.go

```diff
@@ -0,0 +1,203 @@
+	assert.Equal(t, "test_api_key", proxiedRequests[0].Header.Get("DD-API-KEY"))
+
+	assert.Equal(t, "mysubdomain.datadoghq.eu", proxiedRequests[1].Host)
+	assert.Equal(t, "mysubdomain.datadoghq.eu", proxiedRequests[1].URL.Host)
+	assert.Equal(t, "/mypath/mysubpath", proxiedRequests[1].URL.Path)
+	assert.Equal(t, "arg=test", proxiedRequests[1].URL.RawQuery)
+	assert.Equal(t, "test_api_key_1", proxiedRequests[1].Header.Get("DD-API-KEY"))
+
+	assert.Equal(t, "mysubdomain.datadoghq.eu", proxiedRequests[2].Host)
+	assert.Equal(t, "mysubdomain.datadoghq.eu", proxiedRequests[2].URL.Host)
+	assert.Equal(t, "test_api_key_2", proxiedRequests[2].Header.Get("DD-API-KEY"))
+
+	assert.Equal(t, "", loggerOut)
+}
+
+func TestEvpIntakeReverseProxyHandlerInvalidSubdomain(t *testing.T) {
+	conf := newTestReceiverConfig()
+	conf.Site = "us3.datadoghq.com"
+	conf.Endpoints[0].APIKey = "test_api_key"
+
+	request := httptest.NewRequest("POST", "/google.com%3Fattack=/mypath/mysubpath", nil)
+	proxiedRequests, response, loggerOut := sendRequestThroughHandler(conf, request)
+
+	require.Equal(t, 0, len(proxiedRequests))
+	require.Equal(t, http.StatusBadGateway, response.StatusCode, "Got: ", fmt.Sprint(response.StatusCode))
+	require.Contains(t, loggerOut, "invalid subdomain")
+}
+
+func TestEvpIntakeReverseProxyHandlerInvalidPath(t *testing.T) {
+	conf := newTestReceiverConfig()
+	conf.Site = "us3.datadoghq.com"
+	conf.Endpoints[0].APIKey = "test_api_key"
+
+	request := httptest.NewRequest("POST", "/mysubdomain/mypath/my%20subpath", nil)
+	proxiedRequests, response, loggerOut := sendRequestThroughHandler(conf, request)
+
+	require.Equal(t, 0, len(proxiedRequests))
+	require.Equal(t, http.StatusBadGateway, response.StatusCode, "Got: ", fmt.Sprint(response.StatusCode))
+	require.Contains(t, loggerOut, "invalid target path")
+}
+
+func TestEvpIntakeReverseProxyHandlerInvalidQuery(t *testing.T) {
+	conf := newTestReceiverConfig()
+	conf.Site = "us3.datadoghq.com"
+	conf.Endpoints[0].APIKey = "test_api_key"
+
+	request := httptest.NewRequest("POST", "/mysubdomain/mypath/mysubpath?test=bad%20arg", nil)
+	proxiedRequests, response, loggerOut := sendRequestThroughHandler(conf, request)
+
+	require.Equal(t, 0, len(proxiedRequests))
+	require.Equal(t, http.StatusBadGateway, response.StatusCode, "Got: ", fmt.Sprint(response.StatusCode))
+	require.Contains(t, loggerOut, "invalid query string")
+}
+
+func TestEvpIntakeEndpointsFromConfigOverride(t *testing.T) {
+	conf := newTestReceiverConfig()
+	conf.Site = "us3.datadoghq.com"
+	conf.Endpoints[0].APIKey = "test_api_key"
+	conf.EvpIntakeProxy.DDURL = "override.datadoghq.com"
+	conf.EvpIntakeProxy.APIKey = "override_api_key"
+
+	endpoints := evpIntakeEndpointsFromConfig(conf)
+
+	require.Equal(t, 1, len(endpoints))
+	require.Equal(t, endpoints[0].Host, "override.datadoghq.com")
+	require.Equal(t, endpoints[0].APIKey, "override_api_key")
+}
+
+func TestEvpIntakeHandler(t *testing.T) {
```

Comment:
```suggestion
func TestEVPIntakeHandler(t *testing.T) {
```
Can you please capitalise "EVP" everywhere except when it is the prefix of an unexported identifier, in which case all lower-case is fine.